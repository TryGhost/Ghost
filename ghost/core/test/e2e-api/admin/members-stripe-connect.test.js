const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyContentVersion, stringMatching} = matchers;

describe('Members Stripe Connect API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init();
        await agent.loginAsOwner();
    });

    it('can do auth', async function () {
        await agent
            .get(`members/stripe_connect`)
            .expectStatus(302)
            .expectEmptyBody()
            .matchHeaderSnapshot({
                'content-version': anyContentVersion,
                location: stringMatching(/^https:\/\/connect\.stripe\.com\/oauth\/authorize\?response_type=code&scope=read_write&client_id=/),
                'set-cookie': [
                    stringMatching(/^ghost-admin-api-session=/)
                ]
            });
    });
});
