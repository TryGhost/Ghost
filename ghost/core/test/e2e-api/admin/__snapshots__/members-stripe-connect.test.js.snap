// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members Stripe Connect API can do auth 1: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "289",
  "content-type": "text/plain; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "location": StringMatching /\\^https:\\\\/\\\\/connect\\\\\\.stripe\\\\\\.com\\\\/oauth\\\\/authorize\\\\\\?response_type=code&scope=read_write&client_id=/,
  "set-cookie": Array [
    StringMatching /\\^ghost-admin-api-session=/,
  ],
  "vary": "Accept-Version, Origin, Accept, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
